certbot/compat/filesystem.py

```diff
@@ -0,0 +1,125 @@
+"""Compat module to handle files security on Windows and Linux"""
+from __future__ import absolute_import
+
+import os  # pylint: disable=os-module-forbidden
+import stat
+
+try:
+    import ntsecuritycon  # pylint: disable=import-error
+except ImportError:  # pragma: no cover
+    ntsecuritycon = None  # type: ignore
+try:
+    import win32security  # pylint: disable=import-error
+except ImportError:  # pragma: no cover
+    win32security = None  # type: ignore
+
+
+def chmod(file_path, mode):
+    # type: (str, int) -> None
+    """
+    Apply a POSIX mode on given file_path:
+        * for Linux, the POSIX mode will be directly applied using chmod,
+        * for Windows, the POSIX mode will be translated into a Windows DACL that make sense for
+          Certbot context, and applied to the file using kernel calls.
+    :param str file_path: Path of the file
+    :param int mode: POSIX mode to apply
+    """
+    if not win32security:
+        os.chmod(file_path, mode)
+    else:
+        _apply_win_mode(file_path, mode)
+
+
+def _apply_win_mode(file_path, mode):
+    # Resolve symbolic links
+    if os.path.islink(file_path):
```

Comment:
`os.path.realpath` is broken on some versions of Python 3.x on Windows (see https://bugs.python.org/issue9949)

Reply:
I think it would beneficial in the long term. However, I think it should be not done in this PR, as it adds a significant amount of complexity to also act on `os.path` in `certbot.compat.os`, and to link this logic in the existing Certbot codebase.

Reply:
Done: https://github.com/certbot/certbot/issues/7115

Reply:
Done: https://github.com/certbot/certbot/issues/7115